server/sonar-server/src/main/java/org/sonar/server/es/StickyFacetBuilder.java

```diff
@@ -107,9 +108,14 @@ public FilterAggregationBuilder buildTopFacetAggregation(String fieldName, Strin
 
   public FilterAggregationBuilder addSelectedItemsToFacet(String fieldName, String facetName, FilterAggregationBuilder facetTopAggregation, Object... selected) {
     if (selected.length > 0) {
+      String includes = Arrays.stream(selected)
+        .filter(Objects::nonNull)
+        .map(s -> EsUtils.escapeSpecialRegexChars(s.toString()))
+        .collect(Collectors.joining("|"));
```

Comment:
I think collectors can be safely shared when not used with parallel streams. It will save creating a new instance for every query on the index. (previous code had it)